torch/csrc/jit/mobile/function.cpp

```diff
@@ -42,26 +42,43 @@ bool Function::append_operator(
   std::function<void(Stack&)> fn;
 
   auto jit_op = findOperatorFor(opname);
+  std::vector<c10::Argument> args;
   if (jit_op) {
     fn = [jit_op](Stack& stack) { jit_op->getOperation()(&stack); };
+    args = jit_op->schema().arguments();
   } else {
     auto op = c10::Dispatcher::singleton().findSchema(opname_c10);
     if (op.has_value()) {
       fn = [op](Stack& stack) { op->callBoxed(&stack); };
+      if (op->hasSchema()) {
+        args = op->schema().arguments();
+      } else {
+        TORCH_CHECK(false, "arguments are missing for operator ", opname);
+      }
     } else {
       return false;
     }
   }
 
-  if (model_version == 0x3LL &&
-      opname == c10::OperatorName("aten::_convolution", "")) {
-    // Since byte-code versions 0x4L, convolution has an additional
-    // default-value argument (allow_tf32=True, see
-    // https://github.com/pytorch/pytorch/pull/40737). This wrapper handles
-    // backward compatibility with models of byte-code version <= 0x3L, where
-    // this bool argument does not yet exist.
-    fn = [fn](Stack& stack) {
-      stack.push_back(true);
+  // num_specified_args >= 0 indicates number of arguments are available
+  // from model. We can use it to handle backward compatibility.
+  if (num_specified_args >= 0 &&
```

Comment:
Should this be c10::optional instead? Or does it have some perf cost for model loading?

Reply:
Good suggestion! I could use c10::optional to avoid confusion of "-1". I don't this the perf cost for model loading on c10:optional is a big issue.